src/app/zap-templates/zcl/data-model/chip/energy-calendar-cluster.xml

```diff
@@ -0,0 +1,126 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="Energy Management"/>
+
+  <enum name="PeakPeriodSeverityEnum" type="enum8">
+    <cluster code="0x009A"/>
+    <item name="Unused" value="0x00"/>
+    <item name="Low" value="0x01"/>
+    <item name="Medium" value="0x02"/>
+    <item name="High" value="0x03"/>
+  </enum>
+
+  <bitmap name="AuxiliaryLoadBitmap" type="bitmap8">
+    <cluster code="0x009A"/>
+    <field name="AuxiliarySwitch1" mask="0x01"/>
+    <field name="AuxiliarySwitch2" mask="0x02"/>
+    <field name="AuxiliarySwitch3" mask="0x04"/>
+    <field name="AuxiliarySwitch4" mask="0x08"/>
+    <field name="AuxiliarySwitch5" mask="0x10"/>
+    <field name="AuxiliarySwitch6" mask="0x20"/>
+    <field name="AuxiliarySwitch7" mask="0x40"/>
+    <field name="AuxiliarySwitch8" mask="0x80"/>
+  </bitmap>
+
+  <bitmap name="TransitionDayOfWeekBitmap" type="bitmap8">
+    <cluster code="0x009A"/>
+    <field name="Sunday" mask="0x01"/>
+    <field name="Monday" mask="0x02"/>
+    <field name="Tuesday" mask="0x04"/>
+    <field name="Wednesday" mask="0x08"/>
+    <field name="Thursday" mask="0x10"/>
+    <field name="Friday" mask="0x20"/>
+    <field name="Saturday" mask="0x40"/>
+  </bitmap>
+
+  <struct name="Date">
+    <cluster code="0x009A"/>
+    <item fieldId="0" name="Year" type="int16u" default="0" isNullable="true" optional="true"/>
+    <item fieldId="1" name="Month" type="int8u" min="1" max="12" default="1" isNullable="true" optional="true"/>
+    <item fieldId="2" name="Day" type="int8u" min="1" max="31" default="1" isNullable="true" optional="true"/>
+    <item fieldId="3" name="DayOfWeek" type="int8u" min="1" max="7" isNullable="true" optional="true"/>
+  </struct>
+
+  <struct name="CalendarPeriod">
+    <cluster code="0x009A"/>
+    <item fieldId="0" name="StartDate" type="epoch_s" default="0"/>
+    <item fieldId="1" name="Days" type="DayStruct" array="true" min="1"/>
+  </struct>
+
+  <struct name="DayStruct">
+    <cluster code="0x009A"/>
+    <item fieldId="0" name="Date" type="Date" optional="true"/>
+    <item fieldId="1" name="DaysOfWeek" type="TransitionDayOfWeekBitmap" optional="true"/>
+    <item fieldId="2" name="Transitions" type="TransitionStruct" array="true" min="1"/>
+    <item fieldId="3" name="CalendarID" type="int32u" default="0" optional="true"/>
+  </struct>
+
+  <struct name="PeakPeriodStruct">
+    <cluster code="0x009A"/>
+    <item fieldId="0" name="Severity" type="PeakPeriodSeverityEnum"/>
+    <item fieldId="1" name="PeakPeriod" type="int16u"/>
+    <item fieldId="2" name="StartTime" type="epoch_s"/>
+    <item fieldId="3" name="EndTime" type="epoch_s"/>
```

Comment:
These are both nullable in the spec.

I am stopping here; this XML clearly does not match the spec, and the different ordering makes it very hard to review.  Please fix the ordering and make the XML actually match the spec.

Reply:
Fixed by using alchemy tool to generate XML.

Reply:
Perhaps the specification (*.adoc) is changing faster than I can recreate the source and its derivative files.
Can you give me access to the `connectedhomeip-spec` repository?
In addition, verification tasks are actively changing on github itself.
For example, from the last one: now in *.kt files in the description of functions there is a comma after the last parameter.
Was:
```
class GeneralDiagnosticsClusterNetworkFaultChangeEvent(
 val current: List<UInt>,
 val previous: List<UInt>
) {
```
Became:
```
class GeneralDiagnosticsClusterNetworkFaultChangeEvent(
 val current: List<UInt>,
 val previous: List<UInt>,
) {
```

At the initial stage, this was due to the fact that I created XML manually, not knowing about the alchemy tool, but since a week this is no longer relevant.

Reply:
What are you using as a reference if you don't have access to the spec repo?

Reply:
I'd like to echo @cecille's question.

I don't have a way to give you access to the spec repo, but if you need that you should talk to Chris LaPre at the CSA.

> In addition, verification tasks are actively changing on github itself.

Yes, various other changes to the codegen can be happening...  That's normal.

To make this converge faster, please order the XML in the same order as the spec.  Alchemy will try to preserve your existing XML as much as it can if you run it with some XML already in place, so it will not fix issues like the ordering one for you unless you run it from a blank slate....